### PR TITLE
Adding C# Bindings for GDAL

### PR DESCRIPTION
**Pull request by runette**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

This package builds the C# Bindings for the GDAL package.

Although it may seem strange to build these in Conda - the bindings are built in cpp and SWIG using the conda-forge distribution of libgdal and using the same build environment as the libgdal-feedstock. This makes it very simple and natural to do in conda.

The run-time has standard conda-forge dependencies on libgdal and mono and nothing else. The mono versions (OSX and Linux) have an activate script that modifies {DY}LD_LIBRARY_PATH and MONO_PATH env variables so that you can build and run mono applications on the command line in a conda environement and it will just work as youw would expect.

### CODE REVIEW

**Comment by runette** [comment]
@bjornharrtell Please respond to show your willingness to be a maintainer as discussed 

Thanks

(sorry it took so long to add you - been busy)

**Comment by bjornharrtell** [comment]
@runette :+1:

**Review by beckermr — CHANGES_REQUESTED** [review]
Someone familiar with GDAL will need to look at this. 

I noticed that an activation script is setting LD_LIBRARY_PATH and similar for Linux and osx. This is a really bad idea and should not be needed. At minimum the old values should be saved and then restored in the deactivation script.

**Comment by runette** [comment]
> Someone familiar with GDAL will need to look at this.
@beckermr  - do you have a suggestion who? This package does come out of a discussion on the gdal_dev discussion group and has the tacit approval of the core team - [see the discussion](https://lists.osgeo.org/pipermail/gdal-dev/2021-April/053871.html)

> 
> I noticed that an activation script is setting LD_LIBRARY_PATH and similar for Linux and osx. This is a really bad idea and should not be needed. At minimum the old values should be saved and then restored in the deactivation script.

I think that I can get around the LIBRARY_PATH - further experiments show that it was a relic of testing - as long MONO_PATH is set correctly it is not needed. I will submit a new commit fixing that and undoing MONO_PATH when deactivated.

**Comment by runette** [comment]
@beckermr  - all reference to LD_LIBRARY_PATH removed. Deactivate script removes changes to MONO_PATH

File: recipes/Gdal_CSharp/meta.yaml
Comment (chrisburr) [inline recipes/Gdal_CSharp/meta.yaml:2]: Should these be the same version number? I don't see a reason for the `1.0.0` but I might be missing something.

Comment (runette) [reply recipes/Gdal_CSharp/meta.yaml:2]: This was a request from the GDAL team - 3.2.2 is the version number of the GDAL source that is used for the build. They want the version number of the gdal-csharp package to be different from the source version. It allows flexibility and the ability to have out-of-synch bindings releases in the future

Comment (beckermr) [reply recipes/Gdal_CSharp/meta.yaml:2]: This is rather odd as there is no place where we have the mapping of 1.0.0 = 3.2.2 recorded. If the GDAL team wants a separate release cycle, then they need to release the code separately, right?

Comment (beckermr) [reply recipes/Gdal_CSharp/meta.yaml:2]: Also,. the conda-forge bots won't be able to update this feedstock as versions change.

File: recipes/Gdal_CSharp/activate.sh
Comment (beckermr) [inline recipes/Gdal_CSharp/activate.sh:34]: Adding CONDA_PREFIX/lib to the global path is not great. Why is this not adjusting MONO_PATH?

**Comment by runette** [comment]
@chrisburr thank you so much

File: recipes/Gdal_CSharp/activate.sh
Comment (chrisburr) [reply recipes/Gdal_CSharp/activate.sh:34]: I noticed this as well but its setting the variable specified in `$PATH_TYPE` (i.e. `MONO_PATH`)

Comment (beckermr) [reply recipes/Gdal_CSharp/activate.sh:34]: Ahhhh thanks!

Comment (runette) [reply recipes/Gdal_CSharp/activate.sh:34]: It was changing two types of path originally - thus the abstraction. When I removed the library path I did not remove the $PATH_TYPE abstraction.

File: recipes/Gdal_CSharp/meta.yaml
Comment (runette) [reply recipes/Gdal_CSharp/meta.yaml:2]: We will live with manual update tasks for the time being. The long range plan - once this is bedded in - is to separate the source of the bindings into a new repository to make it more maintainable.

**Pull request merged**
